cilium-health/probeserver/main.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+
+	"github.com/cilium/cilium/pkg/health/probeserver"
+
+	flag "github.com/spf13/pflag"
+)
+
+func removePidfile(path string) {
+	if err := os.Remove(path); err != nil && !os.IsNotExist(err) {
+		fmt.Fprintf(os.Stderr, "failed to remove pidfile: %s\n", err.Error())
+	}
+}
+
+func writePidfile(path string) error {
```

Comment:
Reviewer note: I decided to not use the `pkg/pidfile` package here, as I found that it almost doubles the RSS of the process (because it pulls in quite a few other dependencies) to ~13MB.

Reply:
o_o;  Maybe `pkg/pidfile` should go on a diet? If we strip out the logging dependency, does that address this?

Reply:
Okay, I think I mixed up binary size and rss in my notes last week. `pidfile` isn't that bad in terms of memory, it just bloats the binary size:

The current version of the `pkg/pidfile` itself adds 1.8MB of RSS to the responder (from 5.8MB to 7.6MB in total), but adds almost 8MB to the binary size (7.3MB vs 13MB). All the overhead seems all caused by logging. Removing the logging statements brings sizes back down to normal for the responder (i.e. binary size 7.3MB, rss 5.7MB).